posts/2021/10/hpy-0.0.3.md

```diff
@@ -0,0 +1,188 @@
+<!--
+.. title: hpy 0.0.3: Second public release
+.. slug: hpy-0.0.3-second-public-release
+.. date: 2021-10-22 15:00:00 UTC
+.. author: fangerer
+.. tags:
+.. category:
+.. link:
+.. description:
+.. type: text
+-->
+
+HPy 0.0.3 is out! This is already HPy's second officially released version and
+was again made available on PyPI.
+
+The major highlights of this release are new API functions, new debug mode
+features, and Windows support.
+
+<!--TEASER_END-->
+
+What is HPy?
+============
+
+HPy provides a new API for extending Python in C. In other words, you use
+`#include <hpy.h>` instead of `#include <Python.h>`. For more info, look at
+the
+[official documentation](https://docs.hpyproject.org/en/latest/overview.html).
+
+
+Installation
+============
+
+HPy 0.0.3 is best tested on Linux systems but there is also initial support for
+Windows (both `x86_64`).
+For CPython, you need to install it manually, using pip:
+
+```
+$ pip install hpy==0.0.3
+```
+
+!!! Note
+
+    Currently, we provide only the `sdist` (i.e., the `.tar.gz`, no binary
+    wheels). But we plan to ship binary wheels starting with the next release.
+
+
+[PyPy](https://pypy.org) and [GraalPython](https://graalvm.org/python/) already
+come with intrinsic HPy support, so no installation is necessary. HPy 0.0.3 is
+included in the current releases of both, i.e. PyPy 7.3.6 (released on 
+2021-10-17) and GraalPython 21.3.0 (released on 2021-10-19).
+
+To double check the version of HPy which is shipped with those, you can either
+use `pip` or `hpy.universal.get_version()`:
+
+```
+$ pypy -m pip show hpy
+Name: hpy
+Version: 0.0.3
+...
+
+$ graalpython -m pip show hpy
+Name: hpy
+Version: 0.0.3
+...
+
+$ pypy -c 'import hpy.universal; print(hpy.universal.get_version()[0])'
+0.0.3
+
+$ graalpython -c 'import hpy.universal; print(hpy.universal.get_version()[0])'
+0.0.3
+```
+
+API
+===
+
+HPy is still under heavy development and does not yet provide as many API
+functions as Python's C API but more and more pieces are added.
+In this release, we've added following functions:
+
+  - `HPy_Is`
+  - `HPyBytes_FromStringAndSize`
+  - `HPyErr_NewException`
+  - `HPyErr_NewExceptionWithDoc`
+  - `HPyUnicode_AsUTF8AndSize`
+  - `HPyUnicode_DecodeFSDefault`
+  - `HPyImport_ImportModule`
+
+All of the above functions except of `HPy_Is` have equivalents in the C API. You
+just need to remove the leading `H` from the name.
+
+Function `HPy_Is` was added to be able to do object identity comparisons. In the
+C API, you would just do a pointer comparison like this:
+
+```
+PyObject *object_a = /* ... */;
+PyObject *object_b = /* ... */;
+return object_a == object_b;
+```
+
+Since HPy was deliberately designed to prevent such comparisons, you need to use
+the newly introduced function:
```

Comment:
This sentence is a bit obscure. What about this?

```suggestion
In HPy, direct comparisons of handles is forbidden by design, because distinct handles can point to the same object. To compare objects by identity, you need to use `HPy_Is`:
```